feat: allow to reset specific order budgets

daniel-rose opened the pull request.

julianzimmermann commented on bundles/payment-product-restriction/src/FondOfOryx/Zed/PaymentProductRestriction/Business/PaymentMethodFilter/PaymentProductRestrictionPaymentMethodFilter.php:
```diff
@@ -49,18 +48,26 @@ protected function hasRestrictedPaymentMethods(
+            return false;
+        }
+
+        if (empty($this->config->getProductAttributeBlacklistedPaymentMethods())) {
```
combine with if before and do not use empty! https://dev.to/aleksikauppila/using-isset-and-empty-hurts-your-code-aaa

julianzimmermann commented on bundles/payment-product-restriction/src/FondOfOryx/Zed/PaymentProductRestriction/Business/PaymentMethodFilter/PaymentProductRestrictionPaymentMethodFilter.php:
```diff
@@ -78,42 +85,25 @@ protected function removeNotAllowedPaymentMethods(
 
-                    continue;
-                }
+                if (!isset($itemTransfer->getAbstractAttributes()['_'][$productAttributeBlacklistedPaymentMethods])) {
```
see comment before

julianzimmermann commented on bundles/payment-product-restriction/src/FondOfOryx/Zed/PaymentProductRestriction/Business/PaymentMethodFilter/PaymentProductRestrictionPaymentMethodFilter.php:
```diff
@@ -78,42 +84,28 @@ protected function removeNotAllowedPaymentMethods(
-                if ($paymentMethodTransfer->getMethodName() !== $paymentMethodName) {
-                    $filteredPaymentMethods->append($paymentMethodTransfer);
+            foreach ($quoteTransfer->getItems() as $itemTransfer) {
+                $blacklistedPaymentMethodsByProduct = $itemTransfer->getAbstractAttributes()['_'][$productAttributeBlacklistedPaymentMethods];
```
you could make it as const

daniel-rose committed:
feat: persist initial budget also

- add `initial_budget` as column to table `foo_order_budget` and  `foo_order_budget_history`
- add response attribute property `initialBudget` to glue endpoint `order-budget-search`
- increase minor versions for effected bundles `fond-of-oryx/order-budget` and `fond-of-oryx/order-budget-search-rest-api`
- prepare unit tests
- ...

daniel-rose committed:
remove breaking changes

daniel-rose committed:
remove breaking changes

daniel-rose committed:
feat: add `next_initial_budget` for saving `initial_budget` in history before resetting

daniel-rose committed:
chore: add `nextInitialBudget` to the response of `order-budget-search`-endpoint

daniel-rose committed:
feat: allow to reset specific order budgets

daniel-rose committed:
add phpcs fixes

daniel-rose committed:
fix: add required transfer properties

gengjozsef committed:
fix: resolve `unit test` error

daniel-rose committed:
Merge branch 'main' into feature/allow-to-reset-specific-order-budgets
